src/k8s-engine/wsl.ts

```diff
@@ -1537,26 +1537,52 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     return (await this.client?.isServiceReady(namespace, service)) || false;
   }
 
-  requiresRestartReasons(): Promise<Record<string, [any, any] | []>> {
-    if (this.currentAction !== Action.NONE || this.internalState === K8s.State.ERROR || !this.cfg?.enabled) {
-      // If we're in the middle of starting or stopping, we don't need to restart.
-      // If we're in an error state, differences between current and desired could be meaningless
-      // If we aren't running k3s, there are no parameters we care about.
-      return Promise.resolve({});
+  requiresRestartReasons(settings: Settings['kubernetes']): Promise<Record<string, K8s.RestartReason | undefined>> {
+    if (!this.cfg) {
+      return Promise.resolve({}); // No need to restart if nothing exists
     }
 
-    return new Promise((resolve) => {
-      const results: Record<string, [any, any] | []> = {};
-      const cmp = (key: string, actual: number, desired: number) => {
-        results[key] = actual === desired ? [] : [actual, desired];
+    // If we're in the middle of something, or if we're in an error state, there
+    // is no need to tell the use about the need to restart.
+    const quiet = this.currentAction !== Action.NONE || this.internalState === K8s.State.ERROR;
```

Comment:
Looks like most of the code here that doesn't deal with `vmConfig` is the same as in lima.ts -- could it be extracted into a helper function in `k3sHelper.ts`?

Reply:
I'd rather not (until #1997) — it's important that the reasons we find are more tied to the particular backend then the generic code.